lib/pc_patch.c

```diff
@@ -522,6 +522,56 @@ pc_patch_from_patchlist(PCPATCH **palist, int numpatches)
 	return (PCPATCH*)paout;
 }
 
+// first: the first element to select (1-based indexing)
+// count: the number of points to select
+PCPATCH *
+pc_patch_range(const PCPATCH *pa, int first, int count)
+{
+	PCPATCH_UNCOMPRESSED *paout, *pu;
+	uint8_t *buf;
+	size_t size;
+	size_t start;
+
+	assert(pa);
+
+	if ( count < 1 || first < 1 || first > pa->npoints )
+		return NULL;
+
+	if ( first + count > pa->npoints + 1)
+		count = pa->npoints - first + 1;
+
+	paout = pc_patch_uncompressed_make(pa->schema, count);
+	if ( !paout )
+		return NULL;
+	paout->npoints = count;
+
+	pu = (PCPATCH_UNCOMPRESSED *) pc_patch_uncompress(pa);
+	if ( !pu )
+		return NULL;
+
+	buf = paout->data;
+	start = pa->schema->size * (first - 1);
+	size = pa->schema->size * count;
+
+	memcpy(buf, pu->data + start, size);
+
+	if ( ((PCPATCH *) pu) != pa )
+		pc_patch_free((PCPATCH *) pu);
+
+	if ( PC_FAILURE == pc_patch_uncompressed_compute_extent(paout) )
+	{
+		pcerror("%s: extent computation failed", __func__);
```

Comment:
Good catch. And there also is a missing `pc_patch_free(paout)` for the case where `pu == NULL` above. I am fixing those.